src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,133 @@
-""" This is a dummy example """
+import requests
+import pandas as pd
+import time
+import plotly.graph_objects as go
+from plotly.subplots import make_subplots
+
+API_URL = "https://archive-api.open-meteo.com/v1/archive?"
+COORDINATES = {
+    "Madrid": {"latitude": 40.416775, "longitude": -3.703790},
+    "London": {"latitude": 51.507351, "longitude": -0.127758},
+    "Rio": {"latitude": -22.906847, "longitude": -43.172896},
+    }
+VARIABLES = ["temperature_2m_mean", "precipitation_sum", "wind_speed_10m_max"]
+
+def call_api(url:str, cool_off_time:int = 1):
+    """
+    Generic function to call API's and treats possible errors.
+    """
+
+    def rate_limit_handling():
+        """
+        Function to return a response in case of timeout error.
```

Comment:
This docstring is confusing since it doens't match what the function actually does. 

Remember, add docstring sparely only when clarifications add clear value and when you use it, make sure that the comments are clear, concise and that they do match the code.

Missmatches between code and docstrings are super confusing.